### PR TITLE
Added confidential_nodes field to google_container_cluster

### PR DESCRIPTION
**Pull request by melinath**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/7480

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `confidential_nodes` field to `google_container_cluster` resource
```

### CODE REVIEW

**Comment by melinath** [comment]
I'm having some trouble getting the acceptance tests to pass. Specifically, after the "disable" step, the plan is not empty - it seems to not have updated "enabled" to false. Is there something I'm doing obviously wrong? Do you have any advice on how to fix this?

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -623,6 +623,26 @@ func resourceContainerCluster() *schema.Resource {
...
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				MaxItems:    1,
+				Description: `Configuration for the confidential nodes feature, which makes nodes run on confidential VMs. Warning: This configuration can't be changed (or added/removed) after cluster creation without deleting and recreating the entire cluster.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
```
Comment (c2thorn) [inline third_party/terraform/resources/resource_container_cluster.go.erb:638]: I believe this field should be ForceNew as well. Right now the schema says this field is updatable, but there is no logic in the update method that does anything with it. Resulting in no change for your test

Comment (melinath) [reply third_party/terraform/resources/resource_container_cluster.go.erb:638]: Yep, that seems to have been the issue - thanks!

**Review by c2thorn — APPROVED** [review]
LGTM!

**Pull request merged**